### PR TITLE
kms.ListKeys returns CreatedBy/CreatedAt when information is available

### LINKED ISSUE #20132: Update `KMSListKeysHandler` to also return `CreatedAt` and `CreatedBy` fields.
The `KMSListKeysHandler` returns a structure that holds the `CreatedBy` and `CreatedAt` fields, but these are never filled by the [`KMSListKeysHandler`](https://github.com/minio/minio/blob/61bd861a4232ba1c22a9854b94813072bee8bdd8/cmd/kms-handlers.go#L169-L202). Code must be updated, so these fields are properly filled for MinKMS and KES (when possible).

See also https://github.com/minio/madmin-go/issues/289#issuecomment-2245253705.

### PR DESCRIPTION
**Pull request by marktheunissen**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request, I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
Return the metadata about the key in ListKeys, when available.

Fixes https://github.com/minio/minio/issues/20132

## Motivation and Context

> The KMSListKeysHandler returns a structure that holds the CreatedBy and CreatedAt fields, but these are never filled by the KMSListKeysHandler. Code must be updated, so these fields are properly filled for MinKMS and KES (when possible).

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [x] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: internal/kms/kes.go [hunk 1]
```diff
@@ -126,8 +126,16 @@ func (c *kesConn) Status(ctx context.Context) (map[string]madmin.ItemState, erro
...
+func (c *kesConn) ListKeys(ctx context.Context, req *ListRequest) ([]madmin.KMSKeyInfo, string, error) {
+	names, continueAt, err := c.client.ListKeys(ctx, req.Prefix, req.Limit)
+	if err != nil {
+		return nil, "", err
+	}
+	keyInfos := make([]madmin.KMSKeyInfo, len(names))
+	for i := range names {
+		keyInfos[i].Name = names[i]
+	}
+	return keyInfos, continueAt, nil
```
Comment (marktheunissen) [inline internal/kms/kes.go:138]: Manually tested this one on play.min.io

File: internal/kms/kms.go [hunk 2]
```diff
@@ -331,11 +332,13 @@ func (c *kmsConn) ListKeyNames(ctx context.Context, req *ListRequest) ([]string,
 		return nil, "", errListingKeysFailed(err)
 	}
 
-	names := make([]string, 0, len(resp.Items))
-	for _, item := range resp.Items {
-		names = append(names, item.Name)
+	keyInfos := make([]madmin.KMSKeyInfo, len(resp.Items))
+	for i, v := range resp.Items {
+		keyInfos[i].Name = v.Name
+		keyInfos[i].CreatedAt = fmt.Sprintf("%d", v.CreatedAt.Unix())
```
Comment (marktheunissen) [inline internal/kms/kms.go]: I'm using unix timestamp for this format, let me know if it should be something else.

File: internal/kms/kms.go [hunk 3]
```diff
@@ -331,11 +332,13 @@ func (c *kmsConn) ListKeyNames(ctx context.Context, req *ListRequest) ([]string,
...
+		keyInfos[i].CreatedBy = string(v.CreatedBy)
 	}
-	return names, resp.ContinueAt, nil
+	return keyInfos, resp.ContinueAt, nil
```
Comment (marktheunissen) [inline internal/kms/kms.go:340]: Tested this with a local KMS server

File: internal/kms/kms.go (hunk 2 above)
Comment (klauspost) [reply internal/kms/kms.go]: `v.CreatedAt.Format(time.RFC3339)`?

Not really sure why `madmin.KMSKeyInfo.CreatedAt` isn't a time.Time field.

Comment (ramondeklein) [reply internal/kms/kms.go]: Can't we just change this type to a time.Time field? That breaks the API, but this call never returned something before, so I doubt if that would be an issue.

**Review by ramondeklein — APPROVED** [review]
We may consider changing the API for CreatedAt, because using a string is silly.

File: internal/kms/kms.go (hunk 2 above)
Comment (marktheunissen) [reply internal/kms/kms.go]: Yes we can, it just pushes the conversion to a `string` out into the console, which ultimately needs to do so in order to render the JSON response. The PR to make this change is here: https://github.com/minio/console/pull/3421/files

KMSKeyInfo.CreatedAt is a string, but so are all the other `CreatedAt` fields for those KMS structs. To keep them consistent, this PR changes all of them to `time.Time`: https://github.com/minio/madmin-go/pull/296/files

Let me know your thoughts, I think it's good to be consistent in a SDK. It doesn't look like those other structs  are used, and might be dead code? For example I can't find a `/minio/kms/v1/policy` or `/minio/kms/v1/identity` route in the MinIO server, yet madmin has methods for those API calls that return structs like `KMSDescribePolicy`, etc. 

Am I missing something here?

File: go.sum [hunk 4]
```diff
@@ -457,8 +459,6 @@ github.com/minio/kms-go/kes v0.3.0 h1:SU8VGVM/Hk9w1OiSby3OatkcojooUqIdDHl6dtM6Nk
 github.com/minio/kms-go/kes v0.3.0/go.mod h1:w6DeVT878qEOU3nUrYVy1WOT5H1Ig9hbDIh698NYJKY=
 github.com/minio/kms-go/kms v0.4.0 h1:cLPZceEp+05xHotVBaeFJrgL7JcXM4lBy6PU0idkE7I=
 github.com/minio/kms-go/kms v0.4.0/go.mod h1:q12CehiIy2qgBnDKq6Q7wmPi2PHSyRVug5DKp0HAVeE=
-github.com/minio/madmin-go/v3 v3.0.59-0.20240725120704-3cfbffc45f08 h1:VqSGPGX5F5els13e9j/IHN7wZQOj0eM5fvJvMS0E0tA=
-github.com/minio/madmin-go/v3 v3.0.59-0.20240725120704-3cfbffc45f08/go.mod h1:IFAwr0XMrdsLovxAdCcuq/eoL4nRuMVQQv0iubJANQw=
```
Comment (marktheunissen) [inline go.sum]: TODO: remove

**Review by ramondeklein — CHANGES_REQUESTED** [review]
Please update `go.mod` references before merging.

**Review by harshavardhana — APPROVED** [review]

**Comment by harshavardhana** [comment]
Updated the `go.mod` pointing to the latest console master @marktheunissen. Once built, I will merge this.

**Review by poornas — APPROVED** [review]

**Pull request merged**